### PR TITLE
GPU Installer: cgroup Issue and add support for centos and rhel 7.x

### PR DESCRIPTION
**Pull request by rajivml**
* AIFBR-6954
* AIFBR-7201
* Fix for Issue where docker and kubelet are running under different cgroup on Machine restart. 
* Actually it's not a replicated or kubelet issue, when docker is installed by kurl, if we see the contents of docker daemon.json, we will notice this line which ensure that the docker is running under systemd cgroup 
  "exec-opts\": [\"native.cgroupdriver=systemd\"
* when we install docker-nvidia2 plugin and do docker restart, we will notice that the docker-nvidia2 plugin will replace the config under daemon.json with nvidia runtime related config, ideally the plugin should merge the docker config instead of overriding it as a result after the docker restart, docker instead of running under systemd cgroup it will be running under cgroupfs cgroup 

  https://github.com/replicated-collab/uipath-replicated/issues/56
  https://github.com/replicated-collab/uipath-replicated/issues/44

* Now the fix is, instead of we editing daemon.json config, nvidia-docker2 plugin itself can modify the docker config as it require, once it modifies the config , merge the updated config with the config.json which has missing values and restart the docker which will ensure docker is running under systemd cgroup again 

* Added support to install gpu drivers on centos and rhel 7.x 
* Added checks to check if gpu is working or not by running an CUDA container
* Added logic to fetch and print CUDA versions

### CODE REVIEW

File: gpu/install_gpu_drivers.sh [hunk 1]
```diff
@@ -1,92 +1,136 @@
...
-        # update docker daemon json with nvidia runtime
-        set_daemon_json
-        sudo systemctl restart docker
+function install_gpu() {
+    #Install CUDA Drivers for Ubuntu18
+    os=$(. /etc/os-release;echo $ID)
+    distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
+    major_version=$(. /etc/os-release;echo $VERSION_ID | cut -d '.' -f1)
+
+    if [[ "${os}" = "ubuntu" && "${major_version}" = "18" ]];
```
Comment (aniket202) [inline gpu/install_gpu_drivers.sh:9]: We won't support ubuntu 16 as well?

**Review by aniket202 — APPROVED** [review]
LGTM

**Review by mohitseth-nitrr — APPROVED** [review]
LGTM

File: gpu/install_gpu_drivers.sh (hunk 1 above)
Comment (rajivml) [reply gpu/install_gpu_drivers.sh:9]: yeah we don't right now , we have to enhance the script to do so

**Pull request merged**